xapian-core/include/xapian/weight.h

```diff
@@ -588,6 +588,114 @@ class XAPIAN_VISIBILITY_DEFAULT BM25Weight : public Weight {
     double get_sumextra(Xapian::termcount doclen,
 			Xapian::termcount uniqterms) const;
     double get_maxextra() const;
+
+  protected:
+	// BM25+ is disabled.
+	bool bm25_plus = 0;
+};
+
+/** Xapian::BM25PlusWeight subclass implementing the BM25+ weighting function.
+ *
+ * This class implements the BM25+ Probabilistic Weighting scheme which is
+ * described by Lv-Zhai in their paper.  BM25+ is useful when very-long
+ * documents are present in document collection.
+ *
+ * BM25+ uses an extra parameter delta(δ). Formally, BM25PlusWeight(delta)
+ * should be equivalent to BM25Weight(k1, k2, k3, b, min_normlen, delta).
+ */
+class XAPIAN_VISIBILITY_DEFAULT BM25PlusWeight : public BM25Weight {
+	/// Factor to multiply the document length by.
+    mutable Xapian::doclength len_factor;
```

Comment:
Do you actually need to repeat these members, given you're inheriting from `BM25Weight`?